source/UF/H-Levels.lagda

```diff
@@ -33,41 +34,78 @@ open import UF.Subsingletons-FunExt
 open import UF.Subsingletons-Properties
 open import UF.Univalence
 open import UF.UA-FunExt
+open import Notation.General
+open import Naturals.Addition renaming (_+_ to _+'_)
 open import Naturals.Order
 
-module UF.H-Levels (fe : FunExt) (fe' : Fun-Ext) where
+module UF.H-Levels (fe : Fun-Ext)
+                    where
+
+fe' : FunExt
+fe' 𝓤 𝓥 = fe {𝓤} {𝓥}
 
 _is-of-hlevel_ : 𝓤 ̇ → ℕ → 𝓤 ̇
 X is-of-hlevel zero = is-contr X
 X is-of-hlevel succ n = (x x' : X) → (x ＝ x') is-of-hlevel n
 
 hlevel-relation-is-prop : {𝓤 : Universe} (n : ℕ) (X : 𝓤 ̇ )
                         → is-prop (X is-of-hlevel n)
-hlevel-relation-is-prop {𝓤} zero X = being-singleton-is-prop (fe 𝓤 𝓤)
+hlevel-relation-is-prop {𝓤} zero X = being-singleton-is-prop fe
 hlevel-relation-is-prop {𝓤} (succ n) X =
-  Π₂-is-prop fe'
-             (λ x x' → hlevel-relation-is-prop n (x ＝ x'))
+  Π₂-is-prop fe (λ x x' → hlevel-relation-is-prop n (x ＝ x'))
+
+map_is-of-hlevel_ : {X : 𝓤 ̇} {Y : 𝓥 ̇} → (f : X → Y) → ℕ → 𝓤 ⊔ 𝓥 ̇
+map f is-of-hlevel n = each-fiber-of f (λ - → - is-of-hlevel n)
+
+\end{code}
+
+Being of hlevel one is equivalent to being a proposition.
+
+\begin{code}
+
+is-prop' : (X : 𝓤 ̇) → 𝓤  ̇
+is-prop' X = X is-of-hlevel (succ zero)
+
+being-prop'-is-prop : (X : 𝓤 ̇) → is-prop (is-prop' X)
+being-prop'-is-prop X = hlevel-relation-is-prop (succ zero) X
+
+is-prop-implies-is-prop' : {X : 𝓤 ̇} → is-prop X → is-prop' X
+is-prop-implies-is-prop' X-is-prop x x' =
+  pointed-props-are-singletons (X-is-prop x x') (props-are-sets X-is-prop)
+
+is-prop'-implies-is-prop : {X : 𝓤 ̇} → is-prop' X → is-prop X
+is-prop'-implies-is-prop X-is-prop' x x' = center (X-is-prop' x x')
+
+is-prop-equiv-is-prop' : {𝓤 : Universe} {X :  𝓤 ̇} → is-prop X ≃ is-prop' X
```

Comment:
Ahh I don't need it listed but I do need to implicilty supply it.

Reply:
You didn't give the error message, but I'm assuming it's only because you have `{𝓤}` as an argument to `is-prop-equiv-is-prop'`, but you can simply it when you delete it from the type signature. The rest of that code doesn't mention `𝓤` at all.

Reply:
> Ahh I don't need it listed but I do need to implicilty supply it.

I don't think so. None of the code mentions `𝓤`, so why would you need it? Please fix the indentation to be a single space by the way.

Reply:
Oh, I see what you mean now. Yes, you need to implicitly supply it because `X` is also implicit. Sorry for the noise.